### ISSUE TITLE
White border appearing around drop down menus on Linux

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#39069]
### Description

After the update to version 1.67.115 on Linux, there is a white border that appears around/behind browser drop down menus, such as the main menu or bookmark folders, etc.

![image](https://github.com/brave/brave-browser/assets/3299098/994acdc7-73b8-4e49-845d-94476a95c982)

### Steps to reproduce

1. Launch Brave while using Linux/Kubuntu/i3
2. Click hamburger menu or bookmarks bar folder
3. Resulting drop-down menu has a white border around the shadow gradient

### Actual result

White border/background appears

### Expected result

Should be no boarder/background behind drop down menus

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

v1.67.115

### Channel information

- [X] release (stable)
- [X] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

Original user report:
https://community.brave.com/t/white-border-around-dropdown-menus-navigation-menu/553579

**Comment by suryaanshah** [comment]
can confirm the same on v1.67.115 LinuxMint 
![brave-menu-error](https://github.com/brave/brave-browser/assets/47082682/96f37eeb-9811-4f98-b34b-0b63398e36a7)

**Comment by dmitrykvasnikov** [comment]
The same on Arch / i3 (GTK themed) with v1.67.116
![screenshot](https://ibb.co/Bqnn73T)

**Comment by pythys** [comment]
I also confirm this bug.

![2024-06-17_00-45](https://github.com/brave/brave-browser/assets/13056493/65150fcd-4de2-43f9-b1cb-2f585c3b3a44)

**Comment by Utsunom1ya** [comment]
![2024-06-17_05-10](https://github.com/brave/brave-browser/assets/170748007/a97ce77c-462b-4410-9220-e9aea279c096)
![2024-06-17_05-17](https://github.com/brave/brave-browser/assets/170748007/2bed0042-9665-433c-ac9b-5dff7e39a5a2)

Garuda/Archlinux with Hyprland.
6.9.4-zen1-1-zen kernel.

**Comment by GoWeasel** [comment]
> I also confirm this bug. Arch linux on i3wm

- same on Manjaro XFCE
- - brave-browser  1.67.116-1 [Installed])
- - 6.9.3-3 Kernel

**Comment by rmdhnreza** [comment]
Chrome issue [40937108](https://issues.chromium.org/40937108)

TL;DR: On X11, you need a compositor to draw shadows. `xcompmgr` should be enough.

![brave-1](https://github.com/brave/brave-browser/assets/23427492/7f51ad8c-3bdb-43c6-a245-b82234fe5c03)
![brave-2](https://github.com/brave/brave-browser/assets/23427492/c7799ff4-73c1-492f-869c-9368dc647f84)

**Comment by harridu** [comment]
Actually I don't think I need a compositor, losing control over the background of my desktop and introducing other weird side effects. What I do need is some checkbox in Brave to turn off a graphical gimmick I didn't ask for. 

Please don't make the eye-candy mandatory. There is no such problem for Google Chrome, for example. Thank you

**Comment by pythys** [comment]
@harridu  I think the problem actually originates in Chromium and not necessarily in Brave. It seems to be just more visible in Brave. When I open the menu in dark mode in Chromium I still see some "weird" white that is just more tiny. Take a look below, do you see the white corners? That's possibly the same shadow bug.

![2024-06-17_15-06_1](https://github.com/brave/brave-browser/assets/13056493/6a0984eb-6b10-4978-b48d-84fc12e60c1d)

**Comment by inicula** [comment]
Why wasn't this a problem before 1.67? Did this type of menu not have shadows or what?

**Comment by harridu** [comment]
You are right, I didn't notice the white corners. It is not as distracting as Brave's giant white frame. Maybe Brave could reduce the white frame?

I would prefer to skip all that eye-candy, but I understand that this would be an enhancement request to chromium.

**Comment by pythys** [comment]
@harridu Fixing the border might be fixing a mistake with another mistake. The root cause is shadow rendering and I would imagine that's what the brave team should attend to.

If you are annoyed from this bug (as I am) then downgrading is a simple quick solution. I downgraded to version 125.1.66.120 and as the screenshot below shows things are pretty peachy at the moment.
![2024-06-17_17-08](https://github.com/brave/brave-browser/assets/13056493/f2a331f7-a337-4088-9ec9-973315ef9e13)

**Comment by Cthulu201** [comment]
I agree we shouldn't be required to have a compositor, but xcompmgr looks better than picom on my Arch/i3 setup. FWIW in the interim.

**Comment by egbastos666** [comment]
I completely disagree that composite usage should be mandatory. That's a complete useless "eye-candy" feature that brings nothing to the end user rather than something "beautiful". There are a lot of linux users that really doesn't care about this kind of things, that uses i3wm, that uses fluxbox exactly because they are simple and use minimal resources and we don't want to run an additional component just to "looks pretty". I think that the best way to handle that is to include  a flag on chromium that makes possible to disable composite usage run without any shadow or this kind of things

**Comment by pythys** [comment]
@egbastos666 I would suggest that introducing a compositor is a discussion in Chromium. Brave as a derivative product is different flavor of Chromium but it _is_ Chromium.

I'm also not sure how "easy" would it be to override chromium in that regard, but I would imagine a much easier solution is to solve whatever needs solving in Chromium.

**Comment by Cthulu201** [comment]
I'm seeing a black border in Brave now. Chromium doesn't seem to need a compositor on my Linux box:
Version 126.0.6478.114 (Official Build) Arch Linux (64-bit)

**Comment by martinkirk** [comment]
Same problem here.

Linux: MX-23.3_x64 KDE
Brave: Version 1.67.123 Chromium: 126.0.6478.126 (Official Build)

It only happens when compositor is disabled.  

I've haven't noticed the same issue with any other apps, specifically, Chromium  works just fine with the compositor disabled.

**Comment by egbastos666** [comment]
In fact it needs but the side effect when you don't have composite enabled
on your environment is just a small white corner:


[image: image.png]

**Comment by floscr** [comment]
Same issue here when running without a compositor.

I agree that it shouldn't be necessary to run compositor for context menus though, 
as you might disable for the performance overhead.

**Comment by harridu** [comment]
Hi folks,

FYI, I have reported this as a chromium issue, see

	https://issues.chromium.org/issues/347724917

If you've got an account, then please make yourself heard
on this page (pro or cons).

Thank you
Harri

**Comment by Asizdhung** [comment]
Hey I don’t know what the f i have done. I was just trying to download
brave for ios 15 version 1.66.2

**Comment by rubens07** [comment]
Hello,
I fixed this issue by going into the settings in `brave://flags` and setting the `Preferred Ozone platform` option to `Wayland`

![swappy-20240710_082735](https://github.com/brave/brave-browser/assets/57196049/de8f11c7-e5fd-42ef-99c7-d8d9ace90d0a)


My system;
- Arch Linux x86_64 with Hyprland
- Brave 1.67.123 Chromium: 126.0.6478.126 - 64 bits

**Comment by martinkirk** [comment]
> Hello, I fixed this issue by going into the settings in `brave://flags` and setting the `Preferred Ozone platform` option to `Wayland`
> 

This made no difference on my setup, the problem still exists.

Linux: MX-23.3_x64 KDE
Brave: Version 1.67.123 Chromium: 126.0.6478.126 (Official Build)

**Comment by UltraDoge** [comment]
they never fix that

**Comment by fmarier** [comment]
It seems like it's a larger issue that just Chromium. I get the same thing in Nautilus, the GNOME file manager, on i3 + xorg if I don't have `xcompmgr` running:
![Capture d’écran du 2024-07-29 16-34-50](https://github.com/user-attachments/assets/e795eb29-81a4-4845-bde0-bf0ede744d2f)

**Comment by fmarier** [comment]
That said, [some fixes](https://issues.chromium.org/issues/40937108#comment29) are coming in [Chromium 128](https://chromiumdash.appspot.com/commits?commit=60fd468d7c2f2233a893c95ef3c42ee9532659d7&platform=Linux).

**Comment by Zutatensuppe** [comment]
> It seems like it's a larger issue that just Chromium. I get the same thing in Nautilus, the GNOME file manager, on i3 + xorg if I don't have xcompmgr running:

It does not affect nautilus for me, only brave.

**Comment by tonysultana** [comment]
I have the same problem on Debian Bookworm with Openbox and compton for the compositor.

**Comment by Guillex387** [comment]
> ### Description
> After the update to version 1.67.115 on Linux, there is a white border that appears around/behind browser drop down menus, such as the main menu or bookmark folders, etc.
> 
> ![image](https://private-user-images.githubusercontent.com/3299098/339900053-994acdc7-73b8-4e49-845d-94476a95c982.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0MzIxOTQsIm5iZiI6MTcyMzQzMTg5NCwicGF0aCI6Ii8zMjk5MDk4LzMzOTkwMDA1My05OTRhY2RjNy03M2I4LTRlNDktODQ1ZC05NDQ3NmE5NWM5ODIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMDMwNDU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmMzN2ZjNDI4NDJkN2ViZjM5MDUzMWFhODlmNDMyOGU0ODJlNDk5ZjUxOTk4YzI3ODRkYjg3YWJhNGNkNWRhYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zmXB0j3hnPNIZ0jrx4HFVPdw-PtDgnpa3CVU0NTdn14)
> 
> ### Steps to reproduce
> 1. Launch Brave while using Linux/Kubuntu/i3
> 2. Click hamburger menu or bookmarks bar folder
> 3. Resulting drop-down menu has a white border around the shadow gradient
> 
> ### Actual result
> White border/background appears
> 
> ### Expected result
> Should be no boarder/background behind drop down menus
> 
> ### Reproduces how often
> Easily reproduced
> 
> ### Brave version (brave://version info)
> v1.67.115
> 
> ### Channel information
> * [x]  release (stable)
> * [x]  beta
> * [ ]  nightly
> 
> ### Reproducibility
> * [ ]  with Brave Shields disabled
> * [ ]  with Brave Rewards disabled
> * [ ]  in the latest version of Chrome
> 
> ### Miscellaneous information
> Original user report: https://community.brave.com/t/white-border-around-dropdown-menus-navigation-menu/553579

I have the same problem in my Arch linux with Xorg and Qtile, but in the last update of nightly (1.70.62) I don't have the bug

**Comment by benjamind10** [comment]
I have the exact same issue with Ubuntu 24.04 i3wm and can confirm this on two different devices.

**Comment by jneidel** [comment]
Can confirm that it's fixed in latest (nightly) as reported by @Guillex387

Arch + Xorg + i3 + compositor disabled
- 1.68.128 -> border exists
- 1.70.65 -> no border

**Comment by benjamind10** [comment]
I can confirm also fixed on nightly version 1.70.66, I did not have to do any additional steps.

**Comment by Snxrre** [comment]
Fixed for me on version 1:1.69.160-1 (newest version of brave-bin in the AUR)

**Comment by harridu** [comment]
new version is fine for me

**Comment by rebron** [comment]
Closing as fixed in 1.69.153 and higher. Our latest version is 1.69.162 released today.

**Comment by kjozwiak** [comment]
Closing the above as we've received several confirmations that the above has been resolved 👍

**Comment by tonysultana** [comment]
I am updated and still have the border issue.

Version 1.69.162 Chromium: 128.0.6613.120 (Official Build) (64-bit)

[image: image.png]
<https://brave.com/latest/>
Tony

**Comment by fmarier** [comment]
@tonysultana Looks like your screenshot didn't make it through GitHub's email reply.

**Comment by tonysultana** [comment]
It was just a picture of a pop-menu with that heavy transparent border
around it.
